### ISSUE TITLE
Font doesn't render properly, appears as squares (pango 1:1.44.1-1 breaks font)

### ISSUE THREAD

**Issue by D-Vaillant** [sunaku/tamzen-font#25]
I'm not certain that this is an issue specific to just this font but it's the only one that's been affected from what I can tell.

Other .bdf fonts in the same location render properly, Tamzen appears in `fc-list`. The font was working fine until I updated some packages through AUR.

**Comment by nature174** [comment]
Is the font installed on the client?

**Comment by D-Vaillant** [comment]
Edited the OP. The font was previously installed properly and should still be installed properly (it's in /usr/share/fonts, I reran fc-cache, other fonts work properly.)

**Comment by D-Vaillant** [comment]
Okay, I figured it out. Downgrading from pango 1:1.44.1-1 to 1:1.43.0-2 made the font start rendering properly again, so there's definitely something that breaks the font going on there.

**Comment by nature174** [comment]
Report that to the pango website.

**Comment by ccfb3ee765a58cae** [comment]
Tamzen should probably provide bitmap OTFs now that this change has taken place. Seems like the only way forward with the GNOME developer's unnecessary rigidity on this matter.

**Comment by mr-straw** [comment]
Bringing back the otb format (dropped in Tamzen-1.11.4) would fix this issue.

**Comment by leshow** [comment]
Also bit by this, the I don't think that pango considers this a bug and so they aren't going to 'fix' it.

tamzen still renders messed up for me wherever pango is used

**Comment by lcpz** [comment]
> Bringing back the otb format (dropped in Tamzen-1.11.4) would fix this issue.

Unfortunately, the result is ugly. This is the OTB rendering of medium style in sizes 10, 13 and 20:

![tamzen-otb-10r](https://user-images.githubusercontent.com/1777696/68476707-6633c080-0223-11ea-8f27-f7d3f8561b68.png)
![tamzen-otb-13r](https://user-images.githubusercontent.com/1777696/68476708-6633c080-0223-11ea-9115-ed4417721e09.png)
![tamzen-otb-20r](https://user-images.githubusercontent.com/1777696/68476709-6633c080-0223-11ea-9498-a331014a3b20.png)

Seems that size 13 is the template and every other is just a down/upper scale.

I am not a font expert, but I've [made a fork](https://github.com/lcpz/tamzen-font) for Arch Linux. I hope this could be the start of a full port.

**Comment by j-n-f** [comment]
Arch users can work around this as follows

```bash
yay -s fonttosfnt
```

move all the .bdf files into a directory, and run the following script

```sh
#!/bin/sh
for file in *.bdf; do
    NAME=$(echo $file | sed -e 's/\.bdf//g')
    echo $file
    echo $NAME
    fonttosfnt -v -o $NAME.otb -- $file
done
```

Now you have the .otb files. Install as usual (copy to the correct folder, then run `fc-cache -f`)

I don't really have time to figure out the build system for this project, so if someone else wants to do that and put together the pull request I've at least verified that it can be done.

**Comment by j-n-f** [comment]
I would add that presently it looks like Alacritty can render these otb files correctly, but pango doesn't. It looks like pango is adding 1 pixel more between characters than it should. *sigh*

**Comment by sunaku** [comment]
Thanks @j-n-f , I've applied your suggestion in [the `issue-25` branch]( https://github.com/sunaku/tamzen-font/tree/issue-25 ) and also added the generated `*.otb` files there (under `otb/` folder).  Please try it out and let me know if you find any problems.

**Comment by pandasauce** [comment]
I have since then found that fonts traced with BitFonter and converted to TTF in FontLab with some manual adjustments to metrics (they don't carry over) give pixel-perfect results. It's not cheap though...

**Comment by sunaku** [comment]
@D-Vaillant could you help verify? :innocent:  There is only Pango 1.42 available in the distro I use (Void Linux).

**Comment by D-Vaillant** [comment]
I’ve been using it and it works so far; I can check version specifics in a bit. LightDM isn’t liking the font still but I’ll figure out why.

**Comment by sunaku** [comment]
Thanks for confirming.  I also tested this locally and found that the old Pango 1.42 supports it too. :heavy_check_mark: 

These fonts are now merged into master at commit 7477526d941c25962c30a2493cf55557cc18e987.  Closing.
